### PR TITLE
Adds hair tying from BS12

### PR DESCRIPTION
**Pull request by Mwahahahaha**
Adds in hair tying for human species based on the length var.

I'm not sure how this will affect hair tugging. As it currently stands, it changes the h_style. Be free to crack on with that bc i suck or somthing idk

### CODE REVIEW

**Comment by Mwahahahaha** [comment]
!review

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
+		var/datum/sprite_accessory/hair/hair_style = hair_styles_list[h_style]
+		var/selected_string
+		if(hair_style.length <= 1)
+			to_chat(src, "<span class ='warning'>Your hair isn't long enough to tie.</span>")
+			return
+		else
+			var/list/datum/sprite_accessory/hair/valid_hairstyles = list()
+			for(var/hair_string in hair_styles_list)
+				var/list/datum/sprite_accessory/hair/test = hair_styles_list[hair_string]
+				if(test.length >= 2 && "Human" in test.species_allowed)
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]:
```suggestion
				if(test.length >= 2 && ("Human" in test.species_allowed))
```

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+		var/datum/sprite_accessory/hair/hair_style = hair_styles_list[h_style]
+		var/selected_string
+		if(hair_style.length <= 1)
+			to_chat(src, "<span class ='warning'>Your hair isn't long enough to tie.</span>")
+			return
+		else
+			var/list/datum/sprite_accessory/hair/valid_hairstyles = list()
+			for(var/hair_string in hair_styles_list)
+				var/list/datum/sprite_accessory/hair/test = hair_styles_list[hair_string]
+				if(test.length >= 2 && ("Human" in test.species_allowed))
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]: I wonder if this is also shit code. Could be `(species.name in test.species_allowed)` So we could really easily add it to Tajarans and the like. :thinking:

Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]: let's do this instead:
```suggestion
				if(test.length >= 2 && (species.name in test.species_allowed))
```

This way you can make it usable for other species by simply giving them the verb. Like Tajarans.

Comment (Mwahahahaha) [reply code/modules/mob/living/carbon/human/human_powers.dm]: That would probably require some working with adding in length vars for the Tajaran hairstyles too, which all seem to use their inherited value of 1 (they're unset).
Might have to touch in with @LadyFowls regarding hair tugging and whoever else is involved with Tajaran lore regarding hair fashion.

Comment (Mwahahahaha) [reply code/modules/mob/living/carbon/human/human_powers.dm]: I should also probably make mention that the code on Polaris and BS12 makes use of flags to track tieable hair, without a length variable. I was wondering if doing that would be redundant though when we have another system going for hair length.

Comment (Mwahahahaha) [reply code/modules/mob/living/carbon/human/human_powers.dm]: Otherwise, the changes have been committed.

**Pull request merged**